tests/foreman/api/test_hostcollection.py

```diff
@@ -15,446 +15,457 @@
+    """Create host collections with different limits.
 
-        :BZ: 1325989
-        """
+    :id: 86d9387b-7036-4794-96fd-5a3472dd9160
+
+    :expectedresults: The host collection was successfully created and has
+        appropriate limit.
+
+    :CaseImportance: Critical
+    """
+    for _ in range(5):
+        limit = randint(1, 30)
         host_collection = entities.HostCollection(
-            host=[self.hosts[0]], organization=self.org
+            max_hosts=limit, organization=module_org
         ).create()
-        self.assertEqual(len(host_collection.host), 1)
+        assert host_collection.max_hosts == limit
 
-    @pytest.mark.tier1
-    def test_positive_create_with_hosts(self):
-        """Create a host collection that contains hosts.
 
-        :id: bb8d2b42-9a8b-4c4f-ba0c-c56ae5a7eb1d
+@pytest.mark.parametrize("unlimited", [False, True])
+@pytest.mark.tier1
+def test_positive_create_with_unlimited_hosts(module_org, unlimited):
+    """Create host collection with different values of 'unlimited hosts'
+    parameter.
 
-        :expectedresults: The host collection can be read back, and it
-            references two hosts.
+    :id: d385574e-5794-4442-b6cd-e5ded001d877
 
-        :CaseImportance: Critical
+    :parametrized: yes
 
-        :BZ: 1325989
-        """
-        host_collection = entities.HostCollection(host=self.hosts, organization=self.org).create()
-        self.assertEqual(len(host_collection.host), len(self.hosts))
+    :expectedresults: The host collection was successfully created and has
+        appropriate 'unlimited hosts' parameter value.
 
-    @pytest.mark.tier2
-    def test_positive_add_host(self):
-        """Add a host to host collection.
+    :CaseImportance: Critical
+    """
+    host_collection = entities.HostCollection(
+        max_hosts=1 if not unlimited else None,
+        organization=module_org,
+        unlimited_hosts=unlimited,
+    ).create()
+    assert host_collection.unlimited_hosts == unlimited
 
-        :id: da8bc901-7ac8-4029-bb62-af21aa4d3a88
 
-        :expectedresults: Host was added to the host collection.
+@pytest.mark.tier1
+def test_positive_create_with_host(module_org, fake_hosts):
+    """Create a host collection that contains a host.
 
-        :CaseLevel: Integration
+    :id: 9dc0ad72-58c2-4079-b1ca-2c4373472f0f
 
-        :BZ:1325989
-        """
-        host_collection = entities.HostCollection(organization=self.org).create()
-        host_collection.host = [self.hosts[0]]
-        host_collection = host_collection.update(['host'])
-        self.assertEqual(len(host_collection.host), 1)
+    :expectedresults: The host collection can be read back, and it includes
+        one host.
 
-    @pytest.mark.upgrade
-    @pytest.mark.tier2
-    def test_positive_add_hosts(self):
-        """Add hosts to host collection.
+    :CaseImportance: Critical
 
-        :id: f76b4db1-ccd5-47ab-be15-8c7d91d03b22
+    :BZ: 1325989
+    """
+    hosts = fake_hosts
```

Comment:
you shouldn't have to alias fake_hosts. instead, try using it directly.

Reply:
Changes made